File: ch2/exo2.py
#!/usr/bin/env python3

## read a list of numbers
## for each number print the corresponding
## person's name from the list

people = ['fred',
        'betty',
        'barney',
        'dino',
        'wilma',
        'pebbles',
        'bamm-bamm']

print("Enter a list of numbers: ")

indices = []
while True:
    try:
        indices.append(int(input()))
    except:
        break

print()
for idx in indices:
    if idx < 1 or idx > len(indices): continue

    print("{}".format(people[idx - 1]))

File: ch2/exo2.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;

## read a list of numbers
## for each number print the corresponding
## person's name from the list.

my @person = qw(fred barney dino wilma pebbles bamm-bamm);

print("Enter a list of numbers on separate lines: \n");

chomp(my @indices = <STDIN>);

print "\n";
foreach (@indices) {
	next if ($_ < 1 || $_ > @person);

	print $person[$_ - 1], "\n";
}

	
			
